src/scripts/main.js

```diff
@@ -1,3 +1,190 @@
+const cellList = document.querySelectorAll('TD');
+const startButton = document.querySelector('.start');
+const score = document.querySelector('.game-score');
+const rows = [...field.rows].reduce((res, el, index) => {
+  if (!res[index]) {
+    res[index] = [];
+  };
+
+  [...el.children].forEach((item) => {
+    res[index].push(item);
+  });
+
+  return res;
+}, []);
+const cols = [...field.rows].reduce((res, el) => {
+  [...el.children].forEach((item, index) => {
+    if (!res[index]) {
+      res[index] = [];
+    };
+
+    res[index].push(item);
+  });
+
+  return res;
+}, []);
+
+startButton.addEventListener('click', e => {
+  if (!startButton.toggle) {
+    document.addEventListener('keydown', game);
+    showMessage();
+    start();
+    startButton.innerText = 'Restart';
+    startButton.classList.remove('start');
+    startButton.classList.add('restart');
+    startButton.toggle = !startButton.toggle;
+  } else {
+    document.removeEventListener('keydown', game);
+    document.addEventListener('keydown', game);
+    score.innerText = 0;
+    cellList.forEach(el => (el.innerText = ''));
+    start();
+    showMessage();
+  }
+});
+
+function game(e) {
+  if (e.key === 'ArrowRight' || e.key === 'ArrowLeft') {
+    move(rows, e.key);
+  };
+
+  if (e.key === 'ArrowUp' || e.key === 'ArrowDown') {
+    move(cols, e.key);
+  };
+
+  addNewNumber(rows);
+  cellClassHandler();
+}
+
+function move(arr, direction) {
+  arr.forEach(el => {
+    let clear = getClearArray(el);
+
+    for (let i = clear.length; i > 0; i--) {
+      if (clear[i - 1] && clear[i] === clear[i - 1]) {
+        clear[i] = +clear[i] + +clear[i - 1];
+        clear[i - 1] = '';
+      };
+    }
+
+    clear = clear.filter(item => {
+      if (item) {
+        return item;
+      }
+    });
+
+    switch (direction) {
+      case 'ArrowRight':
+      case 'ArrowDown':
+        for (let index = el.length - 1; index >= 0; index--) {
+          clear.length
+            ? el[index].innerText = clear.pop()
+            : el[index].innerText = '';
+        }
+        break;
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        for (let ind = 0; ind < el.length; ind++) {
+          clear.length
+            ? el[ind].innerText = clear.shift()
+            : el[ind].innerText = '';
+        }
+        break;
+    }
+  });
+
+  cellClassHandler();
+
+  cellList.forEach(el => {
+    if (+el.innerText === 2048) {
+      showMessage('win');
+    }
+  });
+};
+
+function findEmptyCell(arr) {
+  const result = [];
+
+  arr.forEach(el => {
+    el.forEach(cell => {
+      if (!cell.innerText) {
+        result.push(cell);
+      }
+    });
+  });
+
+  return result;
+};
+
+function randomCell(max) {
+  return Math.floor(Math.random() * max);
+};
+
+function addNewNumber(array) {
+  const emptyCells = findEmptyCell(array);
+  const position = randomCell(emptyCells.length);
+  let number = randomCell(10);
+
+  if (number <= 1) {
+    number = 4;
+  } else {
+    number = 2;
+  }
```

Comment:
```suggestion
  number = number <= 1 ? 4 : 2
```
- please also put the '4' and '2' to the const, because it will make the code cleaner